pkg/deviceshifu/deviceshifubase/pushtelemetry.go

```diff
@@ -80,7 +92,12 @@ func pushToHTTPTelemetryCollectionService(message *http.Response, telemetryColle
 }
 
 func pushToShifuTelemetryCollectionService(message *http.Response, request *v1alpha1.TelemetryRequest, targetServerAddress string) error {
-	ctx, cancel := context.WithTimeout(context.TODO(), time.Duration(DeviceTelemetryTimeoutInMS)*time.Millisecond)
+	ctxTimeout := DeviceTelemetryTimeoutInMS
+	// get MinIO request's timeout
+	if request.MinIOSetting != nil && request.MinIOSetting.RequestTimeoutMS != nil {
```

Comment:
If we are calling `pushToShifuTelemetryCollectionService` only in `deviceShifu` we can remove the checking of `request.MinIOSetting != nil` since its checked on line 59

Reply:
we can remove the checking because we will call this func on line 42 and 53

Reply:
@Vacant2333  but isn't line 42 and 53 for other protocol? this check only applies to MinIO